packages/ckeditor5-ui/src/textarea/textareaview.ts

```diff
@@ -0,0 +1,211 @@
+			global.window.requestAnimationFrame( () => this._updateAutoGrowHeight() );
+		} );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public override reset(): void {
+		super.reset();
+
+		this._updateAutoGrowHeight();
+	}
+
+	/**
+	 * Updates the {@link #_height} of the view depending on {@link #minRows}, {@link #maxRows}, and the current content size.
+	 *
+	 * **Note**: This method overrides manual resize done by the user using a handle. It's a known bug.
+	 */
+	private _updateAutoGrowHeight( shouldScroll?: boolean ): void {
+		const viewElement = this.element!;
+		const singleLineContentClone = getTextareaElementClone( viewElement, '1' );
+		const fullTextValueClone = getTextareaElementClone( viewElement, viewElement.value );
+		const singleLineContentStyles = singleLineContentClone.ownerDocument.defaultView!.getComputedStyle( singleLineContentClone );
+
+		const verticalPaddings = parseFloat( singleLineContentStyles.paddingTop ) + parseFloat( singleLineContentStyles.paddingBottom );
+		const borders = getBorderWidths( singleLineContentClone );
+		const lineHeight = parseFloat( singleLineContentStyles.lineHeight );
+		const verticalBorder = borders.top + borders.bottom;
+
+		const singleLineAreaDefaultHeight = new Rect( singleLineContentClone ).height;
+		const numberOfLines = Math.round( ( fullTextValueClone.scrollHeight - verticalPaddings ) / lineHeight );
+
+		const maxHeight = this.maxRows * lineHeight + verticalPaddings + verticalBorder;
+
+		// There's a --ck-ui-component-min-height CSS custom property that enforces min height of the component.
+		// This min-height is relevant only when there's one line of text. Other than that, we can rely on line-height.
+		const minHeight = numberOfLines === 1 ? singleLineAreaDefaultHeight : this.minRows * lineHeight + verticalPaddings + verticalBorder;
+
+		// The size of textarea is controlled by height style instead of rows attribute because event though it is
+		// a more complex solution, it is immune to the layout textarea has been rendered in (gird, flex).
+		this._height = Math.min(
+			Math.max(
+				Math.max( numberOfLines, this.minRows ) * lineHeight + verticalPaddings + verticalBorder,
+				minHeight
+			),
+			maxHeight
+		);
+
+		if ( shouldScroll ) {
+			viewElement.scrollTop = viewElement.scrollHeight;
+		}
+
+		singleLineContentClone.remove();
+		fullTextValueClone.remove();
+
+		this.fire<TextareaViewAutoGrowEvent>( 'autoGrow' );
+	}
+
+	/**
+	 * Validates the {@link #minRows} and {@link #maxRows} properties and warns in the console if the configuration is incorrect.
+	 */
+	private _validateMinMaxRows() {
+		if ( this.minRows > this.maxRows ) {
+			/**
+			 * The minimum number of rows is greater than the maximum number of rows.
+			 *
+			 * @error ui-textarea-view-min-rows-greater-than-max-rows
+			 * @param textareaView The misconfigured textarea view instance.
+			 */
+			console.warn( 'ui-textarea-view-min-rows-greater-than-max-rows', { textareaView: this } );
```

Comment:
TBH, I'd go the other way around and throw instead of patching the misconfiguration.

Reply:
I believe that the only time we will get here is through a typo, I don't believe these values will be taken from another place or dynamically provided. So, throwing works for me. The goal is to not have bugs because `minRows > maxRows` and it glitches somewhere down the road.

Reply:
Decision: Let's throw.